internal/dataprocessor/githubprocessor.go

```diff
@@ -0,0 +1,353 @@
+package dataprocessor
+
+import (
+	"encoding/json"
+	"strconv"
+	"time"
+
+	"github.com/google/go-github/v48/github"
+	"github.com/maplelabs/github-audit/metricformator"
+)
+
+const (
+	COMMIT      = "commit"
+	PULLREQUEST = "pull_request"
+	ISSUE       = "issue"
+	GITHUB      = "github"
+)
+
+// GithubProcessor process data from github APIs
+type GithubProcessor struct {
+	// Repository Name
+	RepoName string
+
+	// Repository URL
+	RepoURL string
+
+	// Current time in milliseconds
+	CurrentTimeInMS int64
+
+	// Metricformator instance to customise processed data
+	MetricFormator *metricformator.MetricFormator
+}
+
+// NewGithubProcessor provides new instance of github api processor
+func NewGithubProcessor(repoName string, repoURL string) GithubProcessor {
+	var gp GithubProcessor
```

Comment:
Instead of creating new var, u can simply do 
return GithubProcessor{RepoName: repoName, ...

Reply:
defining other params also , for time variable etc so added as separate variable declaration